devops/github-actions/prefect-deploy-only-when-files-change-including-docker-build.yaml

```diff
@@ -0,0 +1,71 @@
+---
+name: Build and publish Prefect Deployments & Docker images
+
+"on":
+  push:
+    branches:
+      - main
+
+# Do not grant jobs any permissions by default
+permissions: {}
+
+jobs:
+  deploy_flows:
```

Comment:
Would it make sense to split into 2 jobs? Then we wouldn't be running the auth steps unnecessarily
job 1. find flows that need re-deploying, outputting something cute like "found 1 flow to re-deploy"
job 2. deploy them, maybe with successful step output, too

Reply:
We could do this, but I think it would add unnecessary complexity to the workflow.

Reply:
The step outputs or splitting into 2 jobs?

Reply:
both! This is why i created two different workflow files to optionally include the docker auth & docker build pieces!

We always run the GCP auth step first in our workflow because if it fails, theres no reason to continue to building the image. The auth step happens almost immediately whereas the build takes 1-2 (or more minutes) depending on image size